lib/src/painters/player_wave_painter.dart

```diff
@@ -92,35 +92,32 @@ class PlayerWavePainter extends CustomPainter {
 
   void _drawWave(Size size, Canvas canvas) {
     final length = waveformData.length;
+    final halfWidth = size.width / 2;
+    final halfHeight = size.height / 2;
     if (cachedAudioProgress != audioProgress) {
       pushBack();
     }
     for (int i = 0; i < length; i++) {
-      canvas.drawLine(
-        Offset(
-          i * spacing +
-              dragOffset.dx -
-              totalBackDistance.dx +
-              emptySpace +
-              (waveformType.isFitWidth ? 0 : size.width / 2),
-          size.height / 2 +
-              (showBottom
-                  ? ((waveformData[i] * animValue)) * scaleFactor * scrollScale
-                  : 0),
-        ),
-        Offset(
-          i * spacing +
-              dragOffset.dx -
-              totalBackDistance.dx +
-              emptySpace +
-              (waveformType.isFitWidth ? 0 : size.width / 2),
-          size.height / 2 +
-              (showTop
-                  ? -((waveformData[i] * animValue)) * scaleFactor * scrollScale
-                  : 0),
-        ),
-        i < audioProgress * length ? liveWavePaint : fixedWavePaint,
-      );
+      final dx = i * spacing +
+          dragOffset.dx -
+          totalBackDistance.dx +
+          emptySpace +
+          (waveformType.isFitWidth ? 0 : halfWidth);
+      final bottomDy = halfHeight +
+          (showBottom
+              ? ((waveformData[i] * animValue)) * scaleFactor * scrollScale
+              : 0);
+      final topDy = halfHeight +
+          (showTop
+              ? -((waveformData[i] * animValue)) * scaleFactor * scrollScale
```

Comment:
The calculation `-((waveformData[i] * animValue)) * scaleFactor * scrollScale` is repeated above as well, with only difference of polarity of the final result. Please extract to a common variable.